ext/openssl/openssl.c

```diff
@@ -1018,9 +1099,47 @@ PHP_MINIT_FUNCTION(openssl)
 {
 	char * config_filename;
 
-	le_key = zend_register_list_destructors_ex(php_openssl_pkey_free, NULL, "OpenSSL key", module_number);
-	le_x509 = zend_register_list_destructors_ex(php_openssl_x509_free, NULL, "OpenSSL X.509", module_number);
-	le_csr = zend_register_list_destructors_ex(php_openssl_csr_free, NULL, "OpenSSL X.509 CSR", module_number);
+	zend_class_entry ce;
+	INIT_CLASS_ENTRY(ce, "X509Certificate", class_X509Certificate_methods);
+	x509_certificate_ce = zend_register_internal_class(&ce);
+	x509_certificate_ce->ce_flags |= ZEND_ACC_FINAL | ZEND_ACC_NO_DYNAMIC_PROPERTIES;
+	x509_certificate_ce->create_object = x509_certificate_create_object;
+	x509_certificate_ce->serialize = zend_class_serialize_deny;
+	x509_certificate_ce->unserialize = zend_class_unserialize_deny;
+
+	memcpy(&x509_certificate_object_handlers, &std_object_handlers, sizeof(zend_object_handlers));
+	x509_certificate_object_handlers.offset = XtOffsetOf(x509_certificate_object, std);
+	x509_certificate_object_handlers.free_obj = x509_certificate_free_obj;
+	x509_certificate_object_handlers.get_constructor = x509_certificate_get_constructor;
+	x509_certificate_object_handlers.clone_obj = NULL;
```

Comment:
How is cloning going to be handled? I think we should be be using refs up in 1.1 and dup 1.0 possibly.

Reply:
Cloning is currently disabled. I'm not sure I'll have enough time to incorporate its support in this PR before feature freeze starts (next Tuesday), but I can make a follow-up PR afterwards.

Reply:
I don't think cloning should be part of the migration itself. It's a separate feature that allows doing something the current API doesn't (you can't clone a resource). The only case where we have added cloning support until now are curl handles, because the resource API already exposed an equivalent function (curl_copy_handle).

Reply:
Ok it can be done later.